src/psyclone/tests/dynkern_test.py

```diff
@@ -0,0 +1,304 @@
+# -----------------------------------------------------------------------------
```

Comment:
Should this file be in `tests/domain/lfric`?

Reply:
`DynKern` is in `psyclone/dynamo0p3.py` so I mirrored that. I was assuming that when we move `DynKern` to `domain/lfric` we would also move the tests.

Reply:
OK.